### PR TITLE
Cmake build system 2

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

**Review by vlkale — COMMENTED** [review]
Waiting for other's review and then this should be ready for merging into Kokkos Develop

**Comment by crtrott** [comment]
Ok I think this is good enough to merge for now.

File: Build.md
Comment (masterleinad) [inline Build.md:13]: You only get `*dylib` files for `Mac`, right?

```suggestion
5. Specify the generated shared library in the environment variable KOKKOS_TOOLS_LIBRARY when running your Kokkos-based application. 
```

File: Build.md
Comment (masterleinad) [inline Build.md:22]: Step 3 should be the same for both setups.

File: CMakeLists.txt
Comment (masterleinad) [inline CMakeLists.txt:29]: I t looks weid to have a `message(STATUS)` without any message to print.
```suggestion
message(STATUS "Configuring Kokkos-Tools\n")
```

File: cmake/FindITT.cmake
Comment (masterleinad) [inline cmake/FindITT.cmake:3]:
```suggestion
# expectes find_package() calls to use "ITT" package name as well.
```

**Comment by vlkale** [comment]
@crtrott  OK, I see the changes, including the fixes to the Makefiles made. 


Waiting for another review (from @dalg24) in order to merge.

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
Acknowledging I saw Daniels comment but let us just fix that as a follow-up.

**Pull request merged**
